src/NuGet.Core/NuGet.Protocol.Core.v3/LocalRepositories/LocalV3FindPackageByIdResourceProvider.cs

```diff
@@ -25,22 +25,20 @@ public LocalV3FindPackageByIdResourceProvider()
             INuGetResource resource = null;
 
             Uri uri;
-            if (!Uri.TryCreate(source.PackageSource.Source, UriKind.Absolute, out uri)
-                ||
-                !uri.IsFile)
+            if (!Uri.TryCreate(source.PackageSource.Source, UriKind.Absolute, out uri) || uri.IsFile)
             {
-                return Task.FromResult(Tuple.Create(false, resource));
-            }
+                if (Directory.Exists(source.PackageSource.Source)
```

Comment:
This could throw if the source contains invalid path characters. There should be more validation here since this can now get hit when Uri.TryCreate fails for an unknown reason.